moabb/datasets/erpcore2021.py

```diff
@@ -0,0 +1,905 @@
+
+    def data_path(
+        self, subject, path=None, force_update=False, update_path=None, verbose=None
+    ):
+        """
+        Return the data BIDS paths of a single subject.
+
+        Parameters
+        ----------
+        subject : int
+            The subject number to fetch data for.
+        path : None | str
+            Location of where to look for the data storing location. If None,
+            the environment variable or config parameter MNE_(dataset) is used.
+            If it doesn’t exist, the “~/mne_data” directory is used. If the
+            dataset is not found under the given path, the data
+            will be automatically downloaded to the specified folder.
+        force_update : bool
+            Force update of the dataset even if a local copy exists.
+        update_path : bool | None
+            If True, set the MNE_DATASETS_(dataset)_PATH in mne-python config
+            to the given path.
+            If None, the user is prompted.
+        verbose : bool, str, int, or None
+            If not None, override default verbose level (see mne.verbose()).
+
+        Returns
+        -------
+        list
+            A list containing the BIDSPath object for the subject's data file.
+        """
+        if subject not in self.subject_list:
+            raise ValueError("Invalid subject number")
+
+        # Download and extract the dataset
+        dataset_path = self.download_and_extract(
+            path=path, force_update=force_update, update_path=update_path
+        )
+
+        # Create a BIDSPath object for the subject
+        bids_path = BIDSPath(
+            subject=f"{subject:03d}",
+            task=self.task,
+            suffix="eeg",
+            datatype="eeg",
+            root=dataset_path,
+        )
+
+        subject_paths = [bids_path]
+
+        return subject_paths
+
+    def download_and_extract(self, path=None, force_update=False, update_path=None):
+        """
+        Download and extract the dataset.
+
+        Parameters
+        ----------
+        path : str | None
+            The path to the directory where the dataset should be downloaded.
+            If None, the default directory is used.
+        force_update : bool
+            Force update of the dataset even if a local copy exists.
+        update_path: bool | None
+            If True, set the MNE_DATASETS_(dataset)_PATH in mne-python config
+            to the given path.
+
+        Returns
+        -------
+        path : str
+            The dataset path.
+        """
+        if path is not None:
+            path = Path(path) / DATASET_PARAMS[self.task]["folder_name"]
+        else:
+            # Default path is in the user's home directory under 'mne_data'
+            path = Path.home() / "mne_data" / DATASET_PARAMS[self.task]["folder_name"]
+
+        # Check if the dataset already exists and force_update is False
+        if not force_update and path.exists():
+            return path
```

Comment:
This skips download, even if the directory is empty.
 
For example, if you try to download, interrupt it and try again, you get:
```python
>>> data = ErpCore2021_ERN().get_data([1])
FileNotFoundError: [Errno 2] No such file or directory: '~/mne_data/MNE-erpcoreern2021-data/sub-001/eeg'
```

Reply:
Super yes, I am working on this right now!

Reply:
Can you check again @PierreGtch?

Reply:
Perfect! Now download starts even if the directory is already present